.github/workflows/release.yml

```diff
@@ -0,0 +1,308 @@
+          ninja all -j 3 && ninja install
+          popd
+
+      - name: Copy
+        run: |
+          cp build/install/lib/libviam_rust_utils.a builds/libviam_rust_utils-linux_aarch64.a
+          cp build/install/lib/libviamapi.so.${{ needs.prepare.outputs.version }} builds/libviamapi-linux_aarch64.so
+          cp build/install/lib/libviamsdk.so.${{ needs.prepare.outputs.version }} builds/libviamsdk-linux_aarch64.so
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: builds
+          path: builds
+
+  build_linux:
+    if: github.repository_owner == 'viamrobotics'
+    needs: [prepare]
+    runs-on: [self-hosted, x64]
+    container:
+      image: ghcr.io/cross-rs/${{ matrix.image }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+            platform: linux_x86_64
+            image: x86_64-unknown-linux-gnu:main
+          - target: arm-unknown-linux-gnueabihf
+            platform: linux_armv6l
+            image: arm-unknown-linux-gnueabihf:main
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.prepare.outputs.sha }}
+
+      - name: Install dependencies
+        run: |
+          apt-get update
+          apt-get -y dist-upgrade
+          apt-get -y --no-install-recommends install \
+              build-essential \
+              ca-certificates \
+              curl \
+              doxygen \
+              g++ \
+              gdb \
+              git \
+              gnupg \
+              gpg \
+              less \
+              libboost-all-dev \
+              libc-ares-dev \
+              libre2-dev \
+              libssl-dev \
+              ninja-build \
+              pkg-config \
+              software-properties-common \
+              sudo \
+              wget \
+              zlib1g-dev
+
+          # Add the public key for the llvm repository to get the correct clang version
+          bash -c 'wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|apt-key add -'
+          apt-add-repository -y 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main'
+
+          # Add public key and repository to get cmake 3.25+
+          wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - > /usr/share/keyrings/kitware-archive-keyring.gpg
+          echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' > /etc/apt/sources.list.d/kitware.list
+
+          apt-get update
+          apt-get -y --no-install-recommends install -t llvm-toolchain-focal-15 clang-15 clang-tidy-15
+          apt-get -y install cmake
+
+          mkdir -p ${HOME}/opt/src
+          cd ${HOME}/opt/src && \
+            git clone --recurse-submodules -b v1.52.0 --depth 1 --shallow-submodules https://github.com/grpc/grpc && \
```

Comment:
Why are we building gRPC from source?

Reply:
gRPC does not exist in apt for focal sadly, so we have to build from source.